WordPress/Classes/Services/ReaderPostService.m

```diff
@@ -192,6 +192,17 @@ - (void)fetchPostAtURL:(NSURL *)postURL
     }];
 }
 
+-(void)fetchSubscriptionStatusForPost:(ReaderPost *)post
+                              success:(void (^)(BOOL isSubscribed))success
+                              failure:(void (^)(NSError *))failure
+{
+    ReaderPostServiceRemote *remoteService = [[ReaderPostServiceRemote alloc] initWithWordPressComRestApi:[self apiForRequest]];
```

Comment:
I ended up extracting the subscription related methods from `ReaderPostService `and defining them in a new `FollowCommentService.swift` 👍 

As for injecting the remote service -- I was looking at other Service classes to use as an example, and it seemed like RemoteServices were being instantiated inside the init method, instead of being injected. What do you think?

Reply:
> As for injecting the remote service -- I was looking at other Service classes to use as an example, and it seemed like RemoteServices were being instantiated inside the init method, instead of being injected. What do you think?

This is tricky indeed. I had the same problem recently, I solved this way: https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/Services/ReaderCardService.swift#L20

It's still not good but at least I'm able to inject it through the constructor and mock it if necessary in unit tests. Anyway, the way you did is ok too. :)